### PR TITLE
refactor(orb): convert to class-based datasource

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes:

Refactor the `orb` to a class.

## Context:

#8647

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/orb/index.ts [hunk 1]
```diff
@@ -1,64 +1,58 @@
...
-    dep.homepage = res.homeUrl;
+
+  customRegistrySupport = false;
+
+  defaultRegistryUrls = ['https://circleci.com/'];
+
+  @cache({
+    namespace: `datasource-${OrbDatasource.id}`,
+    key: ({ lookupName }: GetReleasesConfig) => lookupName,
+    ttlMinutes: 15,
```
Comment (JamieMagee) [inline lib/datasource/orb/index.ts:21]: @rarkins Two questions:

I added `datasource-` as a prefix for the namespace to bring it in line with the other datasource caches. Is this okay, or would you prefer not to have the cache miss?

Can I raise the `ttlMinutes` to the default of 30, or shall I keep it as 15?

File: lib/datasource/orb/index.spec.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ describe(getName(), () => {
       httpMock.scope(baseUrl).post('/graphql-unstable').reply(200, {});
       expect(
         await getPkgReleases({
-          datasource,
+          datasource: OrbDatasource.id,
```
Comment (viceice) [inline lib/datasource/orb/index.spec.ts:39]: So here too, add a module scoped `const datasource = OrbDatasource.id` so we have less changes. 🙃

File: lib/datasource/orb/index.ts (hunk 1 above)
Comment (viceice) [reply lib/datasource/orb/index.ts:21]: i think we should use default ttl. good idea to unify cache namespaces for datasources

Comment (JamieMagee) [reply lib/datasource/orb/index.ts]: Okay I'm keeping the `datasource-orb` namespace, and set the ttl to use the default value (30 minutes)

**Review by viceice — APPROVED** [review]

**Pull request merged**